tests/test_diagnostics.py

```
@@ -0,0 +1,906 @@
+    ):
+        assert private_value not in serialized
+    source_used = payload["used"]
+    source_temperature = payload["temperature"]
+    source_packets = payload["packets"]
+    assert isinstance(source_used, float) and math.isnan(source_used)
+    assert isinstance(source_temperature, float) and math.isinf(source_temperature)
+    assert isinstance(source_packets, float) and math.isinf(source_packets)
+
+
+async def test_config_entry_diagnostics_validates_temporal_and_unit_metadata(
+    hass: HomeAssistant, make_config_entry: Any
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `make_config_entry`

(ANN401)